pkg/health/health.go

```diff
@@ -53,23 +61,54 @@ func (h ConsulHealthChecker) WatchNodeService(nodename string, serviceID string,
 			if err != nil {
 				errCh <- err
 			}
+			catalogResults := make([]Result, 0)
+			for _, check := range checks {
+				outResult := consulCheckToResult(*check)
+				// only retain checks if they're for this service, or for the
+				// entire node
+				if outResult.Service != serviceID && outResult.Service != "" {
+					continue
+				}
+				catalogResults = append(catalogResults, outResult)
+			}
+			// GetHealth will fail if there are no kv results
 			kvCheck, err := h.consulStore.GetHealth(nodename, serviceID)
-			if err != nil {
-				errCh <- err
+			pickHealthResult(catalogResults, kvCheck, err, resultCh, errCh)
+		}
+	}
+}
+
+// if there are no health results available will place error in errCh then return
+func pickHealthResult(catalogResults []Result, kvCheck kp.WatchResult, err error, resultCh chan<- Result, errCh chan<- error) {
```

Comment:
Please rename `err` to `kvCheckErr` so it's clear what it's supposed to be (`err` is fairly nondescript)